packages/form/src/hooks/use-field.js

```diff
@@ -13,83 +13,49 @@ import { STORE_NAME } from '../data';
 import { useValidation } from './use-validation';
 
 export const useField = ( {
-	name: fieldName,
-	type,
-	value,
-	validation,
+	fieldName,
 	initialValue,
+	isMultiSelect,
+	validation,
 } ) => {
 	const { name: formName } = useContext( Form.Context );
 	const { setFieldValue } = useDispatch( STORE_NAME );
 	const { error, validateField } = useValidation( { fieldName, validation } );
-	const { value: currentValue } = useSelect(
-		( select ) => select( STORE_NAME ).getFieldData( formName, fieldName ),
+	const fieldValue = useSelect(
+		( select ) => {
+			const defaultValue = isMultiSelect ? [] : '';
+			const value = select( STORE_NAME ).getFieldValue(
+				formName,
+				fieldName
+			);
+
+			return ! isNil( value ) ? value : defaultValue;
+		},
 		[ formName, fieldName ]
 	);
 
-	const onChange = ( event ) => {
-		let newValue = event?.target?.value;
-
-		if ( type === 'checkbox' ) {
-			newValue = event.target.checked
-				? uniq( [ ...( currentValue || [] ), event.target.value ] )
-				: filter(
-						currentValue || [],
-						( v ) => v !== event.target.value
-				  );
-		} else if ( type === 'file' ) {
-			newValue = event.target.files;
-		} else if ( type === 'dropdown' ) {
-			newValue = event;
+	useEffect( () => {
+		if ( initialValue ) {
+			setFieldValue( formName, fieldName, initialValue );
 		}
+	}, [] );
 
-		setFieldValue( formName, fieldName, newValue );
+	const onUpdate = ( value, isSelected ) => {
```

Comment:
Not a huge fan of the second `isSelected` argument - it seems like we're still trying to do too much here.  
Like I mentioned in the previous comment, we should really only care about the value.